java/src/org/openqa/selenium/grid/node/config/NodeOptions.java

```diff
@@ -716,4 +718,12 @@ private void report(Map.Entry<WebDriverInfo, Collection<SessionFactory>> entry)
             entry.getValue().size(),
             entry.getKey().isPresent() ? "Host" : "SM"));
   }
+
+  private String unquote(String input) {
+    int len = input.length();
+    if ((input.charAt(0) == '"') && (input.charAt(len - 1) == '"')) {
+      return new Json().newInput(new StringReader(input)).read(Json.OBJECT_TYPE);
```

Comment:
I was initially just using brute force to remove the bounding quotes and the backslashes escaping embedded quotes, but realized this was far too likely to blow up. This approach should be much safer.